src/Symfony/Component/Form/Tests/Console/Descriptor/AbstractDescriptorTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Tests\Console\Descriptor;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\BufferedOutput;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormRegistry;
+use Symfony\Component\Form\FormRegistryInterface;
+use Symfony\Component\Form\ResolvedFormType;
+use Symfony\Component\Form\ResolvedFormTypeFactory;
+
+abstract class AbstractDescriptorTest extends TestCase
+{
+    /** @dataProvider getDescribeResolvedFormTypeTestData */
+    public function testDescribeResolvedFormType(ResolvedFormType $type, $expectedDescription)
+    {
+        $this->assertDescription($expectedDescription, $type);
+    }
+
+    public function getDescribeResolvedFormTypeTestData()
+    {
+        return $this->getDescriptionTestData(array(
+            'resolved_form_type_1' => new ResolvedFormType(new FormType()),
+        ));
+    }
+
+    /** @dataProvider getDescribeGeneralInformationTestData */
+    public function testDescribeGeneralInformation(FormRegistryInterface $object, $expectedDescription)
+    {
+        $options['form_types'] = array(
+            'Symfony\Component\Form\Extension\Core\Type\FormType' => 'service_1',
+        );
+
+        $this->assertDescription($expectedDescription, $object, $options);
+    }
+
+    public function getDescribeGeneralInformationTestData()
+    {
+        return $this->getDescriptionTestData(array(
+            'general_information_1' => new FormRegistry(array(), new ResolvedFormTypeFactory()),
+        ));
+    }
+
+    abstract protected function getDescriptor();
+
+    abstract protected function getFormat();
+
+    private function assertDescription($expectedDescription, $describedObject, array $options = array())
+    {
+        $output = new BufferedOutput(BufferedOutput::VERBOSITY_NORMAL, true);
+        $io = new SymfonyStyle(new ArrayInput(array()), $output);
+
+        $this->getDescriptor()->describe($io, $describedObject, $options);
+
+        $this->assertEquals(trim($expectedDescription), trim(str_replace(PHP_EOL, "\n", $output->fetch())));
```

Comment:
I have no strong opinion about the whole descriptor layer, it will surely ease integrating other formats indeed. 
Actually what I don't like is the testcase itself, the fact that updating a single fixture requires extra effort especially (e.g. for adding an option or changing the output of a single format): the fact that `assertDescription` is separated from `getDescriptionData` makes that there is no place where you have access to both the fixture filename and the actual output, so you just can't do `file_put_contents('this_single_fixture', $correspondingOutput);` for instance.